src/Symfony/Component/Cache/Adapter/ArrayAdapter.php

```diff
@@ -156,8 +193,110 @@ public function commit()
     /**
      * {@inheritdoc}
      */
-    public function delete(string $key): bool
+    public function clear()
     {
-        return $this->deleteItem($key);
+        $this->values = $this->expiries = [];
+
+        return true;
+    }
+
+    /**
+     * Returns all cached values, with cache miss as null.
+     *
+     * @return array
+     */
+    public function getValues()
+    {
+        if (!$this->storeSerialized) {
+            return $this->values;
+        }
+
+        $values = $this->values;
+        foreach ($values as $k => $v) {
+            if (null === $v || 'N;' === $v) {
+                continue;
+            }
+            if (!\is_string($v) || !isset($v[2]) || ':' !== $v[1]) {
+                $values[$k] = serialize($v);
+            }
+        }
+
+        return $values;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function reset()
+    {
+        $this->clear();
```

Comment:
isn't just doing the same thing as clear without the return better ?

Reply:
We must implement both: clear() for psr-6, reset() for service-contracts.
This is copy/paste from existing current implementations that exist in traits.